system/jlib/jatomic.hpp

```diff
@@ -23,8 +23,15 @@
 #ifdef _WIN32
 inline static void spinPause() { YieldProcessor(); }
 #else
-#include "xmmintrin.h"
+# include "x86intrin.h"
+# if defined(_ARCH_X86_)
```

Comment:
Should the #include be inside the #if?
Are there problems with building on non x86 machines if we include that header?
Does it also work on apple?